### PR TITLE
Filter Update channels older than current version

### PR DESCRIPTION
**Pull request by antonlacon**
This alters the update channels users are able to select in manual updates by removing channels that belong to versions older than what the device is running. A device on LE10 will see the LE10 + LE11 channel, but not LE7 -> LE9, for example.

This exposes `/etc/os-releases`'s VERSION_ID to the addon in order to establish the running version. I'm not sure why oe.VERSION was being set to VERSION_ID and then being changed to VERSION, if present. The implication is that VERSION wasn't always there. It's been in the image building at least 4 years, so it's present in releases that'll see this change.

This relies on the naming scheme of LE's release.json's channels, so a custom channel may bypass it. The filter is simple, comparing a channel named `$something-$version` and checking `$version` against VERSION_ID.

### CODE REVIEW

**Comment by CvH** [comment]
@flubshi ^^^ ?
sry no idea who to ping for that :)

File: resources/lib/modules/updates.py [hunk 1]
```diff
@@ -388,7 +388,9 @@ def get_channels(self):
         log.log(str(self.update_json), log.DEBUG)
         if self.update_json:
             for channel in self.update_json:
-                channels.append(channel)
+                # filter versions older than current
+                if float(oe.VERSION_ID) <= float(channel.split('-')[1]):
```
Comment (flubshi) [inline resources/lib/modules/updates.py:392]: Should we check before, if channel contains a `-`? Maybe there are LE forks or update channels without a dash in channel name?

**Comment by flubshi** [comment]
> @flubshi ^^^ ? sry no idea who to ping for that :)

Did my best to test some corner cases (e.g. enable LE10 channel on LE11 and then using this PR): Everything works as expected: Previous channels are hidden :+1: 

For the code I added only one comment. Apart from that it looks fine to me.

From a logic perspective: I think downgrading LE versions makes no sense, since I think there are no backwards compatible DB migrations. From this perspective this PR makes absolutely sense. 
Another corner case might be switching between different LE compatible forks which use a different versioning (scheme). Not sure if this exists and if we need to consider this edge case.

File: resources/lib/modules/updates.py (hunk 1 above)
Comment (antonlacon) [reply resources/lib/modules/updates.py]: I've made it more paranoid. IndexError raises when there's no hyphen as field [1] doesn't exist. If there is a hyphen, it tests that the result is a number. If it comes across something it doesn't understand, it adds it to the list of choices for the user.

**Comment by chewitt** [comment]
Downgrades have never been supported by Kodi (or LE) so having logic to not-support that is good. I'm not bothered about forks with their own weird version schemes either; that's the author of the weird version scheme's problem not ours :)

**Comment by antonlacon** [comment]
I believe this is done.

**Comment by CvH** [comment]
@flubshi are you okay with that ?

**Comment by flubshi** [comment]
> @flubshi are you okay with that ?

I tested on Raspberry PI and channel "https://test.libreelec.tv". The list of Update channels becomes empty with this PR. Without this PR, all update channels (LibreELEC-11.0 to LibreELEC-7.0) are available. Not sure what is wrong.

File: resources/lib/modules/updates.py [hunk 2]
```diff
@@ -388,7 +388,20 @@ def get_channels(self):
...
+                try:
+                    channel_version = channel.split('-')[1]
+                except IndexError:
+                    channel_version = False
+                if channel_version and channel_version.replace('.','',1).isdigit():
+                    channel_version = float(channel_version)
+                else:
+                    channel_version = False
+                if channel_version:
+                    if float(oe.VERSION_ID) <= channel_verison:
```
Comment (flubshi) [inline resources/lib/modules/updates.py:401]:
```suggestion
                    if float(oe.VERSION_ID) <= channel_version:
```

Comment (flubshi) [reply resources/lib/modules/updates.py:401]: With this it works as expected :)

Comment (antonlacon) [reply resources/lib/modules/updates.py]: Fixed. Thanks!
